packages/veritone-react-common/src/components/FaceEngineOutput/FaceDetectionBox/index.js

```diff
@@ -0,0 +1,229 @@
+import React, { Component } from 'react';
+import { shape, number, string, arrayOf, bool, func } from 'prop-types';
+import classNames from 'classnames';
+import Downshift from 'downshift';
+import Input from 'material-ui/Input';
+import Paper from 'material-ui/Paper';
+import Button from 'material-ui/Button';
+import { MenuItem } from 'material-ui/Menu';
+import Avatar from 'material-ui/Avatar';
+import { Manager, Target, Popper } from 'react-popper';
+
+import { msToReadableString } from 'helpers/time';
+import noAvatar from 'images/no-avatar.png';
+import withMuiThemeProvider from 'helpers/withMuiThemeProvider';
+import styles from './styles.scss';
+
+const renderEntitySearchMenu = ({
+  results,
+  getItemProps,
+  highlightedIndex
+}) => {
+  return results.map((result, index) => {
+    return (
+      <MenuItem
+        key={'menu-entity-' + result.id}
+        component="div"
+        {...getItemProps({
+          item: result,
+          index: index,
+          selected: highlightedIndex === index
+        })}
+      >
+        <Avatar
+          src={result.profileImageUrl ? result.profileImageUrl : noAvatar}
+        />
+        <div className={styles.entityInfo}>
+          <div className={styles.menuEntityName}>{result.name}</div>
+          <div className={styles.menuLibraryName}>{result.libraryName}</div>
+        </div>
+      </MenuItem>
+    );
+  });
+};
+
+@withMuiThemeProvider
+class FaceDetectionBox extends Component {
+  static propTypes = {
+    face: shape({
+      startTimeMs: number,
+      endTimeMs: number,
+      object: shape({
+        label: string,
+        uri: string
+      })
+    }),
+    searchResults: arrayOf(
+      shape({
+        entityName: string,
+        libraryName: string,
+        profileImageUrl: string
+      })
+    ),
+    enableEdit: bool,
+    onUpdateEntity: func,
+    addNewEntity: func,
+    onRemoveFaceDetection: func,
+    onEditFaceDetection: func,
+    onClick: func,
+    onSearchForEntities: func
+  };
+
+  state = {
+    editFaceEntity: false,
+    hovered: false
+  };
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Having to use componentWillReceiveProps is usually a sign you need to use a higher order component to contain the editFaceEntity state.